consultation_analyser/consultations/views/answers.py

```diff
@@ -3,52 +3,274 @@
 
 from django.core.exceptions import ObjectDoesNotExist
 from django.core.paginator import Paginator
+from django.db.models import Count, F, Q, QuerySet, Sum, Value
+from django.db.models.functions import Length, Replace
 from django.http import HttpRequest
 from django.shortcuts import get_object_or_404, redirect, render
 
 from .. import models
 from .decorators import user_can_see_consultation
 
 
+def filter_by_response_and_theme(
+    question: models.Question,
+    respondents: QuerySet,
+    responseid: str | None = None,
+    responsesentiment: list[str] | None = None,
+    themesfilter: list[str] | None = None,
+    themesentiment: list[str] | None = None,
+) -> QuerySet:
+    """filter respondents by response themes"""
+    if responseid:
+        respondents = respondents.filter(
+            answer__id=responseid,
+            answer__question_part__question=question,
+        )
+    if responsesentiment:
+        respondents = respondents.filter(
+            answer__in=models.Answer.objects.filter(
+                sentimentmapping__position=responsesentiment, question_part__question=question
+            )
+        )
+    if themesfilter:
+        respondents = respondents.filter(answer__thememapping__theme__in=themesfilter)
+    if themesentiment:
+        respondents = respondents.filter(answer__thememapping__stance=themesentiment)
```

Comment:
I was getting some funny numbers from duplicates here - I think resolved by doing ```respondents = respondents.filter(answer__thememapping__stance=themesentiment).distinct()``` - poss useful for `themesfilter` above too.